apps/teca_temporal_reduction.in

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python@TECA_PYTHON_VERSION@
+try:
+    from mpi4py import *
+    rank = MPI.COMM_WORLD.Get_rank()
+    n_ranks = MPI.COMM_WORLD.Get_size()
+except ImportError:
+    rank = 0
+    n_ranks = 1
+from teca import *
+import argparse
+import numpy as np
+import sys
+import os
+
+set_stack_trace_on_error()
+set_stack_trace_on_mpi_error()
+
+# parse the command line
+parser = argparse.ArgumentParser(
+    description='Reduce the time axis of a NetcCDF CF2 dataset '
+                'using a predfined interval and reduction operator')
+
+parser.add_argument('--input_regex', type=str, required=True,
+                    help='regex matching the desired set of '
+                         'input NetCDF CF2 files')
+
+parser.add_argument('--arrays', nargs='+', required=True,
+                    help='list of arrays to process.')
+
+parser.add_argument('--output_file', type=str, required=True,
+                    help='file pattern for writing output netcdf '
+                         'files. %%t%% will be replaced by a date/time '
+                         'string or time step. See the teca_cf_writer '
+                         'for more information.')
+
+parser.add_argument('--interval', type=str, default='monthly',
+                    help='interval to reduce the time axis to. One of '
+                         'daily, or monthly (monthly)')
+
+parser.add_argument('--operator', type=str, default='average',
+                    help='reduction operator to use. One of minimum, '
+                         'maximum, or average (average)')
+
+parser.add_argument('--steps_per_file', type=int, default=12,
+                    help='number of time steps to write to each output '
+                         'file. (12)')
+
+parser.add_argument('--n_threads', type=int, default=2,
+                    help='Number of threads to use when stremaing the '
+                         'reduction (2)')
+
+parser.add_argument('--verbose', type=int, default=1,
+                    help='enable verbose mode. (1)')
+
+
+args = parser.parse_args()
```

Comment:
if the parser fails, all mpi ranks print the failure/help message.  Maybe this needs to be wrapped like:
```python
try:
    args = parser.parse_args()
except:
    if rank == 0:
        raise
```